src/test/ui/consts/const-eval/format_args.rs

```diff
@@ -0,0 +1,8 @@
+#![crate_type = "lib"]
+
+const A: std::fmt::Arguments = format_args!("literal");
+
+const B: std::fmt::Arguments = format_args!("{}", 123);
+//~^ ERROR calls in constants are limited to
+//~| ERROR calls in constants are limited to
+//~| ERROR temporary value
```

Comment:
Why are there 3 errors here and not just the one we'd expect? I guess that comes from the extra code for the argument `123`? "temporary value dropped while borrowed" is a bit odd though...

Reply:
> "temporary value dropped while borrowed" is a bit odd though...

It expands to

```rust
::core::fmt::Arguments::new_v1(
    &[""],
    &match (&123,) {
        (arg0,) => [::core::fmt::ArgumentV1::new(
            arg0,
            ::core::fmt::Display::fmt,
        )],
    },
);
```

The `[::core::fmt::ArgumentV1::new(arg0, ::core::fmt::Display::fmt)]` is here borrowed and then dropped before the end of the borrow.

Reply:
ah right that would not be promoted.
Better change the test do
```
const B: () = {
  let _ = format_args!(...);
};
```